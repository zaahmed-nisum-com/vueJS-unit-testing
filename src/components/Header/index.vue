<template>
  <div>
    <h2>Header</h2>
  </div>
</template>

<script>
export default {};
</script>

<style></style>
