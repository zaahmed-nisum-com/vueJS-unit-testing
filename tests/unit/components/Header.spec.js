import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Header from '../../../src/components/Header'

describe('header.vue', () => {

    test('snapshot', () => {
        const wrapper = mount(Header)
        expect(wrapper.find('div')).toMatchSnapshot(true)
    })

    let wrapper;
    beforeEach(() => {
        const localVue = createLocalVue()
        wrapper = shallowMount(Header, {
            localVue
        });
    })

    test('renders a vue instance', () => {
        expect(mount(Header).isVueInstance()).toBe(true);
    });

    test('Checks the data-title', () => {
        expect('Weather App').toMatch('Weather App')
    })
})