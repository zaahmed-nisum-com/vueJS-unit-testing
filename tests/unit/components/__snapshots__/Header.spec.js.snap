// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header.vue snapshot 1`] = `
<div>
  <h2>Header</h2>
</div>
`;
